src/components/Users/CreateUserForm.tsx

```diff
@@ -116,35 +113,26 @@ export default function CreateUserForm({ onSubmitSuccess }: Props) {
     }
   }, [phoneNumber, isWhatsApp, form]);
 
-  const onSubmit = async (data: UserFormValues) => {
-    try {
-      const {
-        res,
-        data: user,
-        error,
-      } = await request(UserApi.create, {
-        body: {
-          ...data,
-          // Omit c_password as it's not needed in the API
-          c_password: undefined,
-        } as unknown as UserBase,
+  const { mutateAsync: createUser } = useMutation({
+    mutationFn: mutate(UserApi.create),
+    onSuccess: (user: UserBase) => {
+      toast.success(t("user_added_successfully"));
+      onSubmitSuccess?.(user!);
+    },
+    onError: (error) => {
+      const errors = (error.cause?.errors as any[]) || [];
+      errors.forEach((err) => {
+        const field = err.loc[0];
+        form.setError(field, { message: err.ctx.error });
       });
+    },
+  });
 
-      if (res?.ok) {
-        Notification.Success({
-          msg: t("user_added_successfully"),
-        });
-        onSubmitSuccess?.(user!);
-      } else {
-        Notification.Error({
-          msg: error?.message ?? t("user_add_error"),
-        });
-      }
-    } catch (error) {
-      Notification.Error({
-        msg: t("user_add_error"),
-      });
-    }
+  const onSubmit = async (data: UserFormValues) => {
+    await createUser({
```

Comment:
i don't see any reason why this needs to be async. we are not awaiting this call anyways.

And for post submission things, we are already using the onSuccess callback, let's stick with `mutate` itself

```suggestion
  const onSubmit = (data: UserFormValues) => {
    createUser({
```

Reply:
> i don't see any reason why this needs to be async. we are not awaiting this call anyways.
> 
> And for post submission things, we are already using the onSuccess callback, let's stick with `mutate` itself

made these changes.